### PR TITLE
Fix inheritance of LocationDelegate

### PR DESCRIPTION
**Pull request by uklotzde**
...including some code cleanup in the base class TableItemDelegate.

### CODE REVIEW

File: src/library/tableitemdelegate.h [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 
-class QTableView;
-class QStyleOptionViewItem;
 
 class TableItemDelegate : public QStyledItemDelegate {
     Q_OBJECT
   public:
     explicit TableItemDelegate(QTableView* pTableView);
-    virtual ~TableItemDelegate();
+    ~TableItemDelegate() override = default;
```
Comment (daschuer) [inline src/library/tableitemdelegate.h]: This is redundant on one hand, but I remember that there was an issue with a clang compiler warning. Is this still the case? Do we have a number of these compiler warnings elsewhere? 
If yes, we should think about a mass fix and drop a note here: https://www.mixxx.org/wiki/doku.php/coding_guidelines

**Review by daschuer — COMMENTED** [review]
The changes are looking good to me.
@ronso0 does this change fix you issue?

File: src/library/tableitemdelegate.h (hunk 1 above)
Comment (uklotzde) [reply src/library/tableitemdelegate.h]: This is not redundant! It verifies that the destructor of the base class is virtual.

Comment (daschuer) [reply src/library/tableitemdelegate.h]: Yes right, but is this important here?
But nevermind, it is a question if code style. We just need to state somewhere if we want that line or not. 

Does this issue a compiler warning, btw?

Comment (uklotzde) [reply src/library/tableitemdelegate.h]: I don't know and don't really care. If this pattern is not desired then simply remove all lines that match the regex "~.*override = default;" in a separate commit.

**Comment by ronso0** [comment]
> The changes are looking good to me.
> @ronso0 does this change fix you issue?

somehow it does, at least the OS style is overwritten.
the font color of selected rows is not applied though. It doesn't matter if the 'location' cell is clicked or not.

before:
image

this PR:
image

File: src/library/tableitemdelegate.h (hunk 1 above)
Comment (daschuer) [reply src/library/tableitemdelegate.h]: I have read a bit what others think. While I originally was the opinion that it is unnecessary noise, this post convinced me that it is actually sensible https://github.com/isocpp/CppCoreGuidelines/issues/721#issuecomment-246466603

I have added a sentence to our style guides: 
> This applies also for default destructors even if it looks noisy.

**Comment by daschuer** [comment]
Thank you @uklotzde for this PR and @ronso0 for the manual tests :-)
LGTM

**Comment by ronso0** [comment]
welcome! thx to @uklotzde for the quick fix.

Manual testing would be a no-brainer if there where CI build artifacts for linux..

**Comment by uklotzde** [comment]
But I have no idea how to correctly propagate the highlighted text color to the delegate. This should be done in the common base class.

**Pull request merged**